### DISCUSSION TITLE
patterns/saga/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#92] [Announcements]
# patterns/saga/

Intent To manage and coordinate distributed transactions across multiple services in a fault-tolerant and reliable manner. Explanation Real-world example Imagine a travel agency...

https://java-design-patterns.com/patterns/saga/

**Comment by siy** [comment]
Saga is an antipattern and has no means to ensure consistency in microservices-based systems.

**Reply by iluwatar** [reply]
It's an eventual consistency model. If a transaction fails, sagas ensure consistency by executing compensating transactions to undo changes made by previous transactions.
